packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_prop/number_prop.rs

```diff
@@ -0,0 +1,262 @@
+                // Use the value and came_from_default of `independent_state`
+                if independent_state.came_from_default {
+                    PropCalcResult::FromDefault(independent_state.value)
+                } else {
+                    PropCalcResult::Calculated(independent_state.value)
+                }
+            }
+            1 => {
+                match &numbers_maths_and_strings[0].value {
+                    PropValue::Number(number_value) => {
+                        if self.propagate_came_from_default
+                            && numbers_maths_and_strings[0].came_from_default
+                        {
+                            // if we are basing it on a single number variable and propagating came_from_default,
+                            // then we propagate came_from_default as well as the value.
+                            PropCalcResult::FromDefault(*number_value)
+                        } else {
+                            PropCalcResult::Calculated(*number_value)
+                        }
+                    }
+                    PropValue::Math(math_value) => {
+                        // if we are basing it on a single math, `came_from_default` will be false
+                        PropCalcResult::Calculated(math_value.to_number())
+                    }
+                    PropValue::String(string_value) => {
+                        let converted_number =
+                            string_value.parse().unwrap_or(prop_type::Number::NAN);
+                        PropCalcResult::Calculated(converted_number)
+                    }
+                    _ => panic!(
+                        "Should get number, math, or string dependency for number, found {:?}",
+                        numbers_maths_and_strings[0].value
+                    ),
+                }
+            }
+            _ => {
+                if numbers_maths_and_strings
+                    .iter()
+                    .any(|prop| matches!(&prop.value, PropValue::Math(_) | PropValue::Number(_)))
+                {
+                    // invalid combination. Haven't implemented number dependency with others
+                    PropCalcResult::Calculated(prop_type::Number::NAN)
+                } else {
+                    // Have multiple string variables. Concatenate the string values into a single string
+
+                    if numbers_maths_and_strings.iter().any(|prop| prop.changed) {
+                        let mut value = String::new();
+
+                        for prop in numbers_maths_and_strings {
+                            match &prop.value {
+                                PropValue::String(string_value) => value += string_value,
+                                _ => {
+                                    panic!(
+                                        "Should get boolean or string for boolean, found {:?}",
```

Comment:
`boolean`? You just just have strings here, right? And this should be `unreachable!`

Reply:
It *should* be unreachable, but since the required data is `Vec<PropView<PropValue>>`, there's no place before this to catch that someone connected an invalid data query that returned another type of prop.

Reply:
I suggest adding a blanket test at line #142

Reply:
done